java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/deploy/KonfluxCommand.java

```diff
@@ -0,0 +1,91 @@
+package com.redhat.hacbs.container.deploy;
+
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+
+import jakarta.enterprise.inject.spi.BeanManager;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import com.redhat.hacbs.container.deploy.git.Git;
+import com.redhat.hacbs.container.results.ResultsUpdater;
+
+import io.quarkus.logging.Log;
+import picocli.CommandLine;
+
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+@CommandLine.Command(name = "create-konflux-source")
+public class KonfluxCommand implements Runnable {
+
+    final BeanManager beanManager;
+    final ResultsUpdater resultsUpdater;
+
+    @CommandLine.Option(names = "--task-run-name")
+    String taskRun;
+
+    @CommandLine.Option(required = true, names = "--source-path")
+    Path sourcePath;
+
+    @CommandLine.Option(required = true, names = "--scm-uri")
+    String scmUri;
+
+    @CommandLine.Option(required = true, names = "--scm-commit")
+    String commit;
+
+    @CommandLine.Option(names = "--image-id")
+    String imageId;
+
+    @ConfigProperty(name = "git.deploy.token")
+    Optional<String> gitToken;
+
+    // If endpoint is null then default GitHub API endpoint is used. Otherwise:
+    // for GitHub, endpoint like https://api.github.com
+    // for GitLib, endpoint like https://gitlab.com
+    @CommandLine.Option(names = "--git-url")
+    String gitURL;
+
+    @CommandLine.Option(names = "--git-identity")
+    String gitIdentity;
+
+    @CommandLine.Option(names = "--git-disable-ssl-verification")
+    boolean gitDisableSSLVerification;
+
+    public KonfluxCommand(BeanManager beanManager,
+            ResultsUpdater resultsUpdater) {
+        this.beanManager = beanManager;
+        this.resultsUpdater = resultsUpdater;
+    }
+
+    public void run() {
+        try {
+            // Konflux files should be in the source code in <root>/.jbs/Containerfile|run-build.sh
+            Log.warnf("Source list files %s", sourcePath.toFile().listFiles());
+            Log.warnf("#### DOCKER:\n" + Files.readString(Path.of(sourcePath.toString(), ".jbs/Containerfile")));
+
+            Git.GitStatus archivedSourceTags = new Git.GitStatus();
+
+            // Save the source first regardless of deployment checks
+            if (isNotEmpty(gitIdentity) && gitToken.isPresent()) {
```

Comment:
If git information is not configured the stage becomes a no-op. This is akin to where the source is saved within the DeployCommand ( https://github.com/redhat-appstudio/jvm-build-service/blob/main/java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/deploy/DeployCommand.java#L151 ) - the difference is the latter will also include any recipe build script post/pre changes.